charts/px/templates/portworx-ds-zerostorage.yaml

```diff
@@ -0,0 +1,254 @@
+#Only if Zerostorage is true. 
+{{- if (.Values.zeroStorage) and (eq .Values.zeroStorage true) -}}
+
+#Setting defaults if they are omitted.
+{{- $dataInterface := .Values.dataInterface | default "none" }}
+{{- $managementInterface := .Values.managementInterface | default "none" }}
+
+{{- $lighthouseToken := .Values.lighthouse.token | default "none" }}
+{{- $envVars := .Values.envVars | default "none" }}
+
+{{- $etcdCredentials := .Values.etcd.credentials | default "none:none" }}
+{{- $etcdCA := .Values.etcd.ca | default "none" }}
+{{- $etcdCert := .Values.etcd.cert | default "none" }}
+{{- $etcdKey := .Values.etcd.key | default "none" }}
+{{- $etcdACL := .Values.etcd.acl | default "none" }}
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: portworx-zerostorage
+  namespace: kube-system
+spec:
+  minReadySeconds: 0
+  
+  {{- if eq (.Values.deploymentType | upper | lower) "oci" }}
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  
+  {{- else if eq (.Values.deploymentType | upper | lower) "docker" }}
+  updateStrategy:
+    type: OnDelete
+  
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app:  portworx-zerostorage
```

Comment:
Let's deprecate the zerostorage. Moving forward, we don't need to install on the master. So the entire zerostorage spec isn't needed.